docs/feature_leader_key.md

```diff
@@ -41,6 +41,46 @@ As you can see, you have a few functions. You can use `SEQ_ONE_KEY` for single-k
 
 Each of these accepts one or more keycodes as arguments. This is an important point: You can use keycodes from **any layer on your keyboard**. That layer would need to be active for the leader macro to fire, obviously.
 
+## Leader_new_key
+Alternatively, the leader key also offers a hook to inspect the leader sequence as it grows. This allows for short circuiting when a prefix is matched.
```

Comment:
Please expand this paragraph to provide more detail about `leader_new_key()`:

- When is it called?
- How it it envisioned to be used?
- How does the included example use it?

Reply:
How about something like:

> The function `leader_new_key` is called on every keypress during leading mode. To see the sequence of keypresses, inspect the first `leader_sequence_size` elements of `leader_sequence`. If you wish to exit leader mode for any reason, setting `leading` to `false` inside this function will accomplish that end.

> For example, the following will check `leader_sequence` against a set of macros as the user types, and will detect, run, and set `leading = false` immediately upon the final keypress:

Reply:
That's better!